xdsl/frontend/code_generation.py

```diff
@@ -71,6 +72,87 @@ def visit_Assign(self, node: ast.Assign) -> None:
         # TODO: Implement assignemnt in the next patch.
         pass
 
+    def visit_BinOp(self, node: ast.BinOp):
+        op_name: str = node.op.__class__.__name__
+
+        # Local table which maps the name of a binry operator to the
+        # corresponding Python function.
+        op_to_python = {
+            "Add": "__add__",
+            "BitAnd": "__and__",
+            "LShift": "__lshift__",
+            "Mult": "__mul__",
+            "RShift": "__rshift__",
+            "Sub": "__sub__",
```

Comment:
This list should be a complete list of all things possible in Python. If a specific function isn't yet supported by a frontend types, that should be the error, not "Unknown binary operation"